rai/core_test/node.cpp

```diff
@@ -1426,3 +1426,190 @@ TEST (node, vote_replay)
 		ASSERT_GT (400, iterations);
 	}
 }
+
+TEST (node, reconfirm_mitm)
+{
+	rai::system system (24000, 3);
+	auto & sender (*system.nodes [0]);
+	auto & receiver (*system.nodes [1]);
+	auto & rep (*system.nodes [2]);
+	auto sender_wallet (system.wallet (0));
+	auto receiver_wallet (system.wallet (1));
+	auto rep_wallet (system.wallet (2));
+	rai::keypair sender_key;
+	rai::keypair receiver_key;
+	sender_wallet->insert_adhoc (sender_key.prv);
+	receiver_wallet->insert_adhoc (receiver_key.prv);
+	rep_wallet->insert_adhoc (rai::test_genesis_key.prv);
+
+	// Genesis -> Sender
+	auto block1 = rep_wallet->send_action (rai::test_genesis_key.pub, sender_key.pub, 10 * rai::Mxrb_ratio);
+	int iterations = 0;
+	while (true)
+	{
+		system.poll ();
+		auto balanceOnSender = sender.balance (sender_key.pub);
+		auto balanceOnReceiver = receiver.balance (sender_key.pub);
+		if (balanceOnSender == 10 * rai::Mxrb_ratio && balanceOnReceiver == 10 * rai::Mxrb_ratio)
+		{
+			break;
+		}
+		ASSERT_GT (400, iterations);
+		iterations++;
+	}
+
+	// Install MITM.
+	receiver.block_processor.stop ();
+	receiver.network.packet_filter = [] (uint8_t* buffer, size_t size)
+	{
+		return true;
```

Comment:
This could be fancier and check the block hash before blocking the packet, but this is fine for this test.